src/zampy/datasets/land_cover.py

```diff
@@ -0,0 +1,249 @@
+                    f"Please check input. Dataset: '{self.name}'\n"
+                    f"Variables: '{variable_names}'"
+                )
+                raise ValueError(msg)
+        files = list((ingest_dir / self.name).glob(f"{self.name}_*.nc"))
+
+        ds = xr.open_mfdataset(files, chunks={"latitude": 200, "longitude": 200})
+        ds = ds.sel(time=slice(time_bounds.start, time_bounds.end))
+        new_grid = xarray_regrid.Grid(
+            north=spatial_bounds.north,
+            east=spatial_bounds.east,
+            south=spatial_bounds.south,
+            west=spatial_bounds.west,
+            resolution_lat=resolution,
+            resolution_lon=resolution,
+        )
+        target_dataset = xarray_regrid.create_regridding_dataset(new_grid)
+
+        ds_regrid = ds.regrid.most_common(target_dataset, time_dim="time", max_mem=1e9)
+
+        return ds_regrid
+
+    def convert(
+        self,
+        ingest_dir: Path,
+        convention: Union[str, Path],
+    ) -> bool:
+        converter.check_convention(convention)
+        ingest_folder = ingest_dir / self.name
+
+        data_file_pattern = "land-cover_LCCS_MAP_*.nc"
+
+        data_files = list(ingest_folder.glob(data_file_pattern))
+
+        for file in data_files:
+            # start conversion process
+            print(f"Start processing file `{file.name}`.")
+            ds = xr.open_dataset(file)
+            ds = converter.convert(ds, dataset=self, convention=convention)
+            # TODO: support derived variables
+            # TODO: other calculations
+            # call ds.compute()
+
+        return True
+
+
+def unzip_raw_to_netcdf(
+    ingest_folder: Path,
+    file: Path,
+    overwrite: bool = False,
+) -> None:
+    """Convert a downloaded zip netcdf file to a standard CF/Zampy netCDF file.
+
+    Args:
+        ingest_folder: Folder where the files have to be written to.
+        file: Path to the land cover .zip archive.
+        overwrite: Overwrite all existing files. If False, file that already exist will
+            be skipped.
+    """
+    ncfile = ingest_folder / file.with_suffix(".nc").name
+    if ncfile.exists() and not overwrite:
+        print(f"File '{ncfile.name}' already exists, skipping...")
+    else:
+        ds = extract_netcdf_to_zampy(ingest_folder, file)
+        ds.to_netcdf(path=ncfile)
+
+
+def extract_netcdf_to_zampy(ingest_folder: Path, file: Path) -> xr.Dataset:
+    """Extract zipped data and convert to zampy format.
+
+    Args:
+        ingest_folder: Folder where the files have to be written to.
+        file: Path to the land cover .zip archive.
+
+    Returns:
+        Coarse land cover data satisfying zampy standard.
+    """
+    with ZipFile(file, "r") as zip_object:
+        zipped_file_name = zip_object.namelist()[0]
+        zip_object.extract(zipped_file_name, path=ingest_folder)
+
+    # only keep land cover class variable
+    ds = xr.open_dataset(ingest_folder / zipped_file_name)
```

Comment:
I didn't find an in-memory loading solution for netcdf with `xarray` that similar to `MemoryFile` thingy with `rasterio`, therefore it looks not very elegant. File is unzipped and saved on disk and it will be removed after the nc file is processed. Just wonder if anyone knows some better solutions? @BSchilperoort

Reply:
netCDF has to be on disk to be read, so there is no way around that. The neatest way might be using the `tempfile` module.

`tempfile.TemporaryDirectory` can be used as context manager, and the temporary dir + files will be cleaned up upon completion.

Reply:
Good idea! I will try `tempfile` then. Usually use it for unit tests and it is indeed a nice solution here. Thanks 👍.